File: day1/sonar_sweep.js
const fs = require("fs");

let input = fs.readFileSync("./input.txt");

let array = input.toString().split("\n").map((element) => parseInt(element.trim()));
console.log(array);

let counter = 0;
for(let i = 1; i < array.length; i++) {
    if(array[i] > array[i-1]) {
        counter = counter + 1;
    }
}
console.log(counter);

File: day1/sonar_sweep.cpp
#include <bits/stdc++.h>
#include<limits.h>

using namespace std;


int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    freopen("input.txt", "r", stdin);

    int counter = 0;
    vector <int> list;

    int t;
    cin >> t;

    for(int i = 0; i < t; i++) {
        int current_sweep;
        cin >> current_sweep;
        list.push_back(current_sweep);
    }

    for(int i = 1; i < list.size(); i++) {
        if(list[i] > list[i-1]) {
            counter++;
        }
    }
    cout << counter << "\n";

    return 0;
}